main.cpp

```diff
@@ -69,46 +69,497 @@ write 3 UDTs below that EACH have:
+}
+
+Cabin::FlightAttendant::~FlightAttendant()
+{
+    std::cout << "Destruct Cabin::FlightAttendant" << std::endl;
+}
+
+void Cabin::FlightAttendant::dance(float insanityLevel)
+{
+    std::cout << "I had like " << insanityLevel << " cups of coffee before this" << std::endl;
+}
+
+float Cabin::FlightAttendant::bendKnee()
+{
+    earSize *= 2.0f;
+    return earSize;
+}
+
+void Cabin::FlightAttendant::speak()
+{
+    std::cout << "About what?" << std::endl;
+    std::cout << deoderantBrand << " " << favoriteDanceType << std::endl;
+}
+
+float Cabin::FlightAttendant::pourCoffee(float coffeeLevel, int numberOfOrders)
+{
+    for(int order = 0; order < numberOfOrders; ++order)
+    {
+        if(coffeeLevel > 0.5f)
+        {
+            std::cout << "Pour cup" << std::endl;
+            coffeeLevel -= 0.5f;  
+        }
+        else
+        {
+            std::cout << "No more coffee" << std::endl;
+            return 0.0f;
+        }
+    }
+    return coffeeLevel;
+}
+
+void Cabin::dimLights(float newLevel)
+{
+    newLevel += legroomDepth;
+    std::cout << snackType << " and " << gingerAletype << std::endl;
+}
+
+float Cabin::serveCoffee(FlightAttendant flightAttendant)
+{
+    flightAttendant.bendKnee();
+    return legroomDepth;
+}
+
+void Cabin::playMusic()
+{
+    if(numberOfPassengers < numberOfSeats)
+    {
+        joseph.numberOfEyeballs = 0;
+        nadine.numberOfElbows = 0;
+    }
+}
+
+float Cabin::reclineSeat(float maxReclineAngle, bool rearPassengerIsHappy)
+{
+    float angle = 0.0f;
 
+    while(angle < maxReclineAngle)
+    {
+        if(angle > 30.0f)
+        {
+            rearPassengerIsHappy = false;
+        }
+
+        if(rearPassengerIsHappy)
+        {
+            std::cout << "Recline seat" << std::endl;
+            angle += 5.0f;
+        }
+        else
+        {
+            std::cout << "No recline" << std::endl;
+            return angle;
+        }
+    }
+    return angle;
+}
 /*
- copied UDT 3:
- */
+copied UDT 3:
+*/
+struct Iphone
+{
+    Iphone();
+
+    std::string operatingSystem;
+    float screenSize{8.0f};
+    std::string speakerType;
+    std::string cameraType;
+    float batteryLife = 100.0f;
+
+    void browseTheWeb(std::string url);
+    void playMusic(std::string songName);
+    void makePhoneCall(int number);
+    void screenShrinker(int magnitude);
+};
 
+Iphone::Iphone():
+operatingSystem("iOS"),
+speakerType("Bose"),
+cameraType("Nikon")
+{   
+    std::cout << "Constructing IPhone" << std::endl;
+}
+
+void Iphone::browseTheWeb(std::string url)
+{   
+    operatingSystem = speakerType + cameraType;
+    std::cout << "www." << url << std::endl;
+}
+
+void Iphone::playMusic(std::string songName)
+{
+    screenSize = batteryLife;
+    std::cout << speakerType << " " << songName << std::endl;
+}
+
+void Iphone::makePhoneCall(int number)
+{
+    std::cout << "Listen on on " << speakerType << " speakers" << std::endl;
+    number += 1;
+}
+
+void Iphone::screenShrinker(int magnitude)
```

Comment:
This function name does not contain a verb/action.  the function name is a noun
- [ ] revise, functions should be named after actions because functions do work, and work is an action